test/test-parse-function.cpp

```diff
@@ -610,13 +622,121 @@ TEST(test_parse, function_statement_without_parameter_list_or_body) {
             ERROR_TYPE_FIELD(
                 error_missing_function_parameter_list, function_name,
                 offsets_matcher(&code, strlen(u8"function "), u8"f")),
+            ERROR_TYPE_FIELD(
+                error_missing_function_body, function_name,
+                offsets_matcher(&code, strlen(u8"function "), u8"f")),
             ERROR_TYPE_FIELD(
                 error_missing_operand_for_operator, where,
                 offsets_matcher(&code, strlen(u8"function f"), u8".")),
             ERROR_TYPE_FIELD(
                 error_missing_semicolon_after_statement, where,
                 offsets_matcher(&code, strlen(u8"function f.x()"), u8""))));
   }
+  {
```

Comment:
Nit: Put these test cases into their own `TEST`:
```suggestion
}

TEST(test_parse, function_statement_without_body) {
  {
```